### PR TITLE
PKCS #11 Unit Tests

### PR DESCRIPTION
**Pull request by lundinc2**
PKCS #11 Unit Tests

Description
-----------
* 100% line coverage.
* Refactored common setup and teardown stubs to macros to reduce LOC.

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c [hunk 1]
```diff
@@ -195,6 +196,18 @@ int suiteTearDown( int numFailures )
 
 /* ==========================  Helper functions  ============================ */
 
+#define prvCommonInitStubs()               \
+    xResult = prvInitializePkcs11();       \
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );  \
+    xResult = prvOpenSession( &xSession ); \
+    TEST_ASSERT_EQUAL( CKR_OK, xResult );  \
```
Comment (alfred2g) [inline libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:203]: extra

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c [hunk 2]
```diff
@@ -940,18 +1092,23 @@ void test_pkcs11_C_CreateObjectECPrivKey( void )
...
+    TEST_ASSERT_EQUAL( CKR_HOST_MEMORY, xResult );
 
-    /* TODO: Remove this. Currently memory is malloc'd by PKCS #11 and free'd in the
-     * MbedTLS stack, but those functions are stubbed out. Need to update this test case
-     * to have a memory life cycle that is properly mocked. */
-    usMallocFreeCalls--;
+    prvCommonDeinitStubs();
+
+    /* Since malloc does not require a NULL check, the stub was called one time too many. */
+    usMallocFreeCalls++;
```
Comment (cobusve) [inline libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:1111]: Please avoid this by letting the malloc stub check for null and not count when it is null.

**Pull request merged**